src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTestTransformer.java

```diff
@@ -51,6 +54,24 @@
 
   private GapicCodePathMapper pathMapper;
 
+  private class TestCaseNameTable {
+    private HashMap<String, Integer> nameCount;
+
+    public TestCaseNameTable() {
+      nameCount = new HashMap<>();
+    }
+
+    public String getTestName(SurfaceNamer namer, Method method) {
+      String testedMethodName = method.getSimpleName();
+      Integer count = 1;
+      if (nameCount.containsKey(testedMethodName)) {
+        count = nameCount.get(testedMethodName) + 1;
+      }
+      nameCount.put(testedMethodName, count);
+      return namer.getTestCaseName(method, count);
+    }
```

Comment:
I think algorithm wise my approach is slightly more efficient (hashmap vs while loop) but there is probably no big difference. If you have a strong preference I could address this in a separate PR.